make.ps1

```diff
@@ -0,0 +1,270 @@
+
+function Check-Scripts-Command
+{
+	if ((Get-Command "luac.exe" -ErrorAction SilentlyContinue) -ne $null)
+	{
+		echo "Testing Lua scripts..."
+		foreach ($script in ls "mods/*/maps/*/*.lua")
+		{
+			luac -p $script
+		}
+		echo "Check completed!"
+	}
+	else
+	{
+		echo "luac.exe could not be found. Please install Lua."
+	}
+}
+
+function Docs-Command
+{
+	if (Test-Path ./engine/OpenRA.Utility.exe)
+	{
+		./engine/OpenRA.Utility.exe all --docs | Out-File -Encoding "UTF8" DOCUMENTATION.md
+		./engine/OpenRA.Utility.exe all --lua-docs | Out-File -Encoding "UTF8" Lua-API.md
+		echo "Docs generated."
+	}
+	else
+	{
+		UtilityNotFound
+	}
+}
+
+function FindMSBuild
+{
+	$msBuildVersions = @("4.0")
+	foreach ($msBuildVersion in $msBuildVersions)
+	{
+		$key = "HKLM:\SOFTWARE\Microsoft\MSBuild\ToolsVersions\{0}" -f $msBuildVersion
+		$property = Get-ItemProperty $key -ErrorAction SilentlyContinue
+		if ($property -eq $null -or $property.MSBuildToolsPath -eq $null)
+		{
+			continue
+		}
+		$path = Join-Path $property.MSBuildToolsPath -ChildPath "MSBuild.exe"
+		if (Test-Path $path)
+		{
+			return $path
+		}
+	}
+	return $null
+}
+
+function UtilityNotFound
+{
+	echo "OpenRA.Utility.exe could not be found. Build the project first using the `"all`" command."
+}
+###############################################################
+############################ Main #############################
+###############################################################
+if ($args.Length -eq 0)
+{
+	echo "Command list:"
+	echo ""
+	echo "  all             Builds the game, its development tools and the mod dlls."
+	echo "  version	        Sets the version strings for all mods to the latest"
+	echo "                  version for the current Git branch."
+	echo "  clean           Removes all built and copied files."
+	echo "                  from the mods and the engine directories."
+	echo "  test            Tests the mod's MiniYAML for errors."
+	echo "  check           Checks .cs files for StyleCop violations."
+	echo "  check-scripts   Checks .lua files for syntax errors."
+	echo "  docs            Generates the trait and Lua API documentation."
+	echo ""
+	$command = (Read-Host "Enter command").Split(' ', 2)
+}
+else
+{
+	$command = $args
+}
+
+#Run the same command on the engine's make file
+if ($command -eq "all" -or $command -eq "clean")
+{
+	if (Test-Path ./engine/OpenRA.sln)
+	{
+		cd ".\engine\"
+		$exp = ".\make.cmd " + $command
+		Invoke-Expression $exp
+		echo ""
+		cd ..
+	}
+	else
+	{
+		if (Get-Command 'git' -ErrorAction SilentlyContinue)
+		{
+			git submodule update --init
+
+			if (Test-Path ./engine/OpenRA.sln)
+			{
+				cd ".\engine\"
+				$exp = ".\make.cmd " + $command
+				Invoke-Expression $exp
+				echo ""
+				cd ..
+			}
+			else
+			{
+				echo "Failed to initialize the submodule. You need to download the engine by hand."
+				return
+			}
+		}
+		else
+		{
+			echo "Unable to locate Git. You need to download the engine by hand."
+			return
```

Comment:
These `return`s mean the window immediately closes and we can't see the message.